core/src/main/java/org/neo4j/ogm/cypher/Filter.java

```diff
@@ -220,14 +246,67 @@ public void setNestedRelationshipEntity(boolean nestedRelationshipEntity) {
 		this.nestedRelationshipEntity = nestedRelationshipEntity;
 	}
 
-    /**
-     * Returns the result of passing the property value through the transformer associated with the comparison operator
-     * on this {@link Filter}.
-     *
-     * @return The transformed property value
-     */
-    public Object getTransformedPropertyValue() {
-       return this.comparisonOperator.getPropertyValueTransformer().transformPropertyValue(this.propertyValue);
-    }
+	/**
+	 * Returns the result of passing the property value through the transformer associated with the comparison operator
+	 * on this {@link Filter}.
+	 *
+	 * @return The transformed property value
+	 */
+	public Object getTransformedPropertyValue() {
+		return this.comparisonOperator.getPropertyValueTransformer().transformPropertyValue(this.value);
+	}
+
+	public FilterFunction getFunction() {
+		return function;
+	}
+
+	public void setFunction(FilterFunction function) {
+		this.function = function;
+	}
+
+	/**
+	 * @param nodeIdentifier
+	 * @param addWhereClause
+	 * @return The filter state as a CYPHER fragment.
+	 */
+	public String toCypher(String nodeIdentifier, boolean addWhereClause) {
+		StringBuilder cypher = new StringBuilder();
+		if (this.function == FilterFunction.NONE) {
+			String uniquePropertyName = isNested() ?
+					getNestedPropertyName() + "_" + getPropertyName() : getPropertyName();
```

Comment:
I think the idea of a comparator is a good one actually, rather than just a bag of keys and values. So I wouldn't change that. In fact I'd consider make the a comparator a formal part of the specification, and provide default implementations for NONE and DISTANCE encapsulated in the enum class. That means that a user could create their own distance comparator and pass it in, or just use the default.